rust/operator-binary/src/controller.rs

```diff
@@ -1416,7 +1460,11 @@ async fn build_node_rolegroup_statefulset(
             service_name: rolegroup_ref.object_name(),
             template: pod_template,
             update_strategy: Some(StatefulSetUpdateStrategy {
-                type_: Some("OnDelete".to_string()),
+                type_: if version_change_state.is_none() {
```

Comment:
I would use a boolean as parameter here like `can_rolling_update` and do the none check when calling this.